docs/kernel/modify_kernel.md

```diff
@@ -2,105 +2,182 @@
 
-By default the CONFIG_MAGIC_SYSRQ setting is disabled.  For this tutorial we will enable it. Using your favorite editor open the config file.  Find the CONFIG_MAGIC_SYSRQ setting, then make the adjustments as shown here:
 ```bash
-# Before
-# CONFIG_MAGIC_SYSRQ is not set
+# Get the hash for the source tar
+SOURCEHASH=$(sha256sum kernel-5.15.102.1.tar.gz | awk '{print $1}')
 
-# After
-CONFIG_MAGIC_SYSRQ=y
-CONFIG_MAGIC_SYSRQ_DEFAULT_ENABLE=0x1
-CONFIG_MAGIC_SYSRQ_SERIAL=y
+# Update the hash for the source tar in the signatures file
+sed -i 's/    "kernel-5.15.102.1.tar.gz": .*/    "kernel-5.15.102.1.tar.gz": "'"$SOURCEHASH"'"/' kernel.signatures.json
 ```
 
-Note that the kernel spec file, from the CBL-Mariner repo, requires implicitly enabled settings to be explicitly set.  In this case enabling CONFIG_MAGIC_SYSRQ is insufficient because CONFIG_MAGIC_SYSRQ_DEFAULT_ENABLE and CONFIG_MAGIC_SYSRQ_SERIAL are implicitly enabled.  If they were missing, compilation of the kernel would fail.  In general, when an error of this nature occurs, the build log file for the kernel will indicate what needs to be changed.  For example, if we _only_ set CONFIG_MAGIC_SYSRQ=y, the build would eventually fail with the build output shown here:
+### Customize a Kernel
 
-```
-time="2021-02-05T11:16:15-08:00" level=debug msg="Magic SysRq key (MAGIC_SYSRQ) [Y/n/?] y"
-time="2021-02-05T11:16:15-08:00" level=debug
-time="2021-02-05T11:16:15-08:00" level=debug msg="Error in reading or end of file."
-time="2021-02-05T11:16:15-08:00" level=debug msg="  Enable magic SysRq key functions by default (MAGIC_SYSRQ_DEFAULT_ENABLE) [0x1] (NEW) "
-time="2021-02-05T11:16:15-08:00" level=debug
-time="2021-02-05T11:16:15-08:00" level=debug msg="Error in reading or end of file."
-time="2021-02-05T11:16:15-08:00" level=debug msg="  Enable magic SysRq key over serial (MAGIC_SYSRQ_SERIAL) [Y/n/?] (NEW) "
-.
-.
-.
-time="2021-02-05T11:16:15-08:00" level=debug msg="+ cat config_diff"
-time="2021-02-05T11:16:15-08:00" level=debug msg="--- new_config\t2021-02-05 19:16:15.316175432 +0000"
-time="2021-02-05T11:16:15-08:00" level=debug msg="+++ current_config\t2021-02-05 19:16:09.440117553 +0000"
-time="2021-02-05T11:16:15-08:00" level=debug msg="@@ -6484,8 +6484,6 @@"
-time="2021-02-05T11:16:15-08:00" level=debug msg=" # end of Compile-time checks and compiler options"
-time="2021-02-05T11:16:15-08:00" level=debug msg=" "
-time="2021-02-05T11:16:15-08:00" level=debug msg=" CONFIG_MAGIC_SYSRQ=y"
-time="2021-02-05T11:16:15-08:00" level=debug msg="-CONFIG_MAGIC_SYSRQ_DEFAULT_ENABLE=0x1"
-time="2021-02-05T11:16:15-08:00" level=debug msg="-CONFIG_MAGIC_SYSRQ_SERIAL=y"
-time="2021-02-05T11:16:15-08:00" level=debug msg=" CONFIG_DEBUG_KERNEL=y"
-time="2021-02-05T11:16:15-08:00" level=debug msg=" CONFIG_DEBUG_MISC=y"
-time="2021-02-05T11:16:15-08:00" level=debug msg=" "
-```
+Now you can  make your modifications to the necessary config files.  
+* For AMD64, modify the `config` file.  
```

Comment:
This really depends on view that you're reading from. Kernel community uses both. RPM documentation uses x86_64 when enumerating architectures. When talking out loud, usually easier to say AMD64 than x86_64.

My vote is `x86_64` since it matches the RPM enumeration, is company-agnostic, and avoids potential confusion with ARM64 when reading fast (which we could debate should be ARM64 or AARCH64 😄 ).

Reply:
Auto-generated config also says `Linux/x86_64` https://github.com/microsoft/CBL-Mariner/blob/c02f10c620cbc2507139655c49fcd57a23a417db/SPECS/kernel/config#L3

Reply:
> This really depends on view that you're reading from. Kernel community uses both. RPM documentation uses x86_64 when enumerating architectures. When talking out loud, usually easier to say AMD64 than x86_64.

I don't think the kernel uses both, in the kernel code AMD64 specifically applies to AMD models (with a few exceptions); the 64-bit x86 instruction set architecture is generically referred to as x86_64. However, it is true that everyone uses the terms interchangeably and will understand 'amd64' (used in a generic context) refers to x86_64. But technically, unless you are using a Debian-based distro (which specifically refers to the x86_64 arch as 'amd64' for historical reasons), x86_64 is the 'correct' term.

https://fedoraproject.org/wiki/Architectures
https://en.opensuse.org/openSUSE:Build_Service_supported_build_targets#Supported_processor_platforms
https://wiki.debian.org/SupportedArchitectures

> 
> My vote is `x86_64` since it matches the RPM enumeration, is company-agnostic, and avoids potential confusion with ARM64 when reading fast (which we could debate should be ARM64 or AARCH64 smile ).

aarch64 is the correct name for the architecture (though the kernel source - but not uname - does use arm64, as do Debian-based distros).

Here are all the kernel architecture names:

```
$ git grep -E '\s*UTS_MACHINE\s*:='
Makefile:UTS_MACHINE    := $(ARCH)
arch/arm64/Makefile:UTS_MACHINE := aarch64_be
arch/arm64/Makefile:UTS_MACHINE := aarch64
arch/loongarch/Makefile:UTS_MACHINE             := loongarch64
arch/mips/Makefile:UTS_MACHINE          := mips
arch/mips/Makefile:UTS_MACHINE          := mips64
arch/parisc/Makefile:UTS_MACHINE        := parisc64
arch/powerpc/Makefile:UTS_MACHINE := $(subst $(space),,$(machine-y))
arch/riscv/Makefile:    UTS_MACHINE := riscv64
arch/riscv/Makefile:    UTS_MACHINE := riscv32
arch/s390/Makefile:UTS_MACHINE  := s390x
arch/sh/Makefile:UTS_MACHINE            := sh
arch/sparc/Makefile:UTS_MACHINE    := sparc
arch/sparc/Makefile:UTS_MACHINE   := sparc64
arch/x86/Makefile:        UTS_MACHINE := i386
arch/x86/Makefile:        UTS_MACHINE := x86_64

```